pkg/resource/container.go

```diff
@@ -0,0 +1,302 @@
+		}
+
+		blobInfo := types.BlobInfo{
+			Digest: layer.Digest,
+			Size:   layer.Size,
+		}
+		layerBlob, _, err := imageSource.GetBlob(ctx, blobInfo, cache)
+		if err != nil {
+			return fmt.Errorf("could not download layer blob: %v", err)
+		}
+		defer layerBlob.Close()
+
+		err = r.decompress(layerBlob, layer.MediaType, layer.Size)
+		if err != nil {
+			return fmt.Errorf("could not decompress layer: %v", err)
+		}
+	}
+	return nil
+
+}
+
+// The decompression process is a little more involved so separated out.
+func (r *Container) decompress(t io.Reader, mediaType string, size int64) error {
+	// The maximum file size should be less than 10x the layer size.
+	size = size * 10
+
+	var tarReader *tar.Reader
+
+	if strings.Contains(strings.ToLower(mediaType), "gzip") {
+		gzReader, err := gzip.NewReader(t)
+		if err != nil {
+			return err
+		}
+		tarReader = tar.NewReader(gzReader)
+		defer gzReader.Close()
+	} else {
+		tarReader = tar.NewReader(t)
+	}
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+		path, err := sanitizePath(r.clonePath, header.Name)
+		if err != nil {
+			logger.Error("%s - skipped", err)
+			continue
+		}
+		info := header.FileInfo()
+		if info.IsDir() {
+			if err = os.MkdirAll(path, 0700); err != nil {
+				return err
+			}
+			continue
+		}
+
+		file, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600) // #nosec G304
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		// Copy a maximum number of bytes (layer size * 10) so we do not get "bombs". It is unlikely that a file
+		// with significant entropy will be compressed more than 10x. We can review this.
+		n, err := io.CopyN(file, tarReader, size)
+		if err != nil && err != io.EOF {
+			return fmt.Errorf("could not copy file to disk: %v", err)
+		}
+		if n >= size {
+			logger.Error("copying file %s did not finish due to max file size: %v", file.Name(), err)
+		}
+	}
+	return nil
+}
+
+// ClonePath returns where this repo has been cloned if cloned else ""
+func (r *Container) ClonePath() string {
+	return r.clonePath
+}
+
+// Depth returns the depth for things that have version control
+func (r *Container) Depth() uint16 {
+	return 0
+}
+
+// SetDepth allows you to adjust the depth for the resource
+func (r *Container) SetDepth(depth uint16) {
+	// no-op
+}
+
+// SetCloneTimeout lets you adjust the timeout before the clone aborts
+func (r *Container) SetCloneTimeout(timeout time.Duration) {
+	// no-op
+}
+
+// Since returns the date after which things should be scanned for things
+// that have versions
+func (r *Container) Since() string {
+	return ""
+}
+
+// ReadFile provides a way to access values in the resource
+func (r *Container) ReadFile(path string) ([]byte, error) {
+	return os.ReadFile(filepath.Join(r.ClonePath(), filepath.Clean(path)))
+}
+
+// Walk traverses the resource like a directory tree
+func (r *Container) Walk(fn WalkFunc) error {
```

Comment:
I was thinking on this. 

The primary place this is used is when processing findings. 
Maybe we should move that to the resources? The below is how i approached this, building on what you have done for git commit messages. 
If here we just sent the resource the findings to do as they wished first, then we would encapsulate that knowledge into the resource instead of elsewhere. Thoughts?

https://github.com/leaktk/scanner/pull/38/files#diff-f8472fe0bafccc33f268c74d1bd2a697594914d4e1d796805b254249cbc78828R181-R192